crates/sui-core/src/epoch/reconfiguration.rs

```diff
@@ -191,28 +191,48 @@ where
         net_config.request_timeout = Some(Duration::from_secs(5));
         net_config.http2_keepalive_interval = Some(Duration::from_secs(5));
 
+        let cur_clients = self.net.load().authority_clients.clone();
+
         for validator in next_epoch_validators {
-            let address = Multiaddr::try_from(validator.net_address).map_err(|e| {
-                SuiError::GenericAuthorityError {
-                    error: e.to_string(),
+            let public_key_bytes = match AuthorityPublicKeyBytes::from_bytes(
+                &validator.pubkey_bytes,
+            ) {
+                Err(err) => {
+                    error!("Error parsing validator public key. Skip this validator in the committee: {:?}", err);
+                    continue;
                 }
-            })?; //TODO: handle what happens if a validator registers with a faulty address
-
-            let channel =
-                net_config
-                    .connect_lazy(&address)
-                    .map_err(|e| SuiError::GenericAuthorityError {
-                        error: e.to_string(),
-                    })?;
-            let client: A = A::recreate(channel);
+                Ok(result) => result,
+            };
+            if let Some(existing_client) = cur_clients.get(&public_key_bytes) {
+                // TODO: We currently do not allow a validator to modify its network settings.
```

Comment:
Do you mean we don't allow them except at epoch change?

Reply:
No, as in we just don't provide a way for people to do so. And since here we use public key to decide whether to recreate the network connection, even if we provide a way for people to do so, the network connection won't change here.